cmake/Modules/FindAubio.cmake

```diff
@@ -1,21 +1,36 @@
-if (NOT USE_SELF_BUILT_AUBIO)
-    find_package(PkgConfig REQUIRED)
-    if(PKG_CONFIG_FOUND)
-        # The version we look for is legacy, may need some fine tunning
-        pkg_check_modules(AUBIO IMPORTED_TARGET GLOBAL aubio>=0.4.9)
-            if(AUBIO_FOUND)
-                add_library(aubio ALIAS PkgConfig::AUBIO)
-            endif()
-    endif()
-endif()
+include(LibFindMacros)
+include(LibFetchMacros)
+
+set(Aubio_GIT_VERSION "14fec3da6749fbcc47b56648d7a38296eccd9499")
 
-if (NOT AUBIO_FOUND)
-    message("Building aubio from sources...")
-    include(FetchContent)
-    FetchContent_Declare(aubio-sources
-      GIT_REPOSITORY https://github.com/performous/aubio.git
-      GIT_TAG        14fec3da6749fbcc47b56648d7a38296eccd9499
-      SOURCE_DIR aubio-src
-    )
-    FetchContent_MakeAvailable(aubio-sources)
+if(SELF_BUILT_AUBIO STREQUAL "ALWAYS")
+	message(STATUS "aubio forced to build from source")
+	libfetch_git_pkg(Aubio
```

Comment:
I don't really get the purpose of the libfetch_git_pkg function, basically you replace:
 ```
  FetchContent_Declare(aubio-sources
      GIT_REPOSITORY https://github.com/performous/aubio.git
      GIT_TAG        14fec3da6749fbcc47b56648d7a38296eccd9499
)
     FetchContent_MakeAvailable(aubio-sources)
```
by ```

libfetch_git_pkg(Aubio
        	REPOSITORY ${SELF_BUILT_GIT_BASE}/aubio.git
		REFERENCE  ${Aubio_GIT_VERSION}
	)
```
what is the expected benefit from it?

Reply:
The goal is:
- having a single way to have a library available (it mostly behave like our FindMacros package (returns the same kind of variables: found, includes, ...)
- hiding all the folder logic (for example the naming of the folder where it is downloaded and things like that),
- discover automatically the file (probably useful for headers only packages)
- hide the fact that FetchContent can fetch anything (this is especially crafter for git)
- avoid FetchContent_Declare/_MakeAvailable logic

Reply:
I'm not really fan of such "syntaxic suggar" wrappers as it make life more complicated to people that know cmake and even to those who try to understand, but if I'm the only one to dislike it, go on...